athena-udfs/src/test/java/com/amazonaws/athena/connectors/udfs/AthenaUDFHandlerTest.java

```diff
@@ -105,29 +107,16 @@ public void testDecompressTruncatedInput()
     @Test
     public void testKmsDecryption() throws Exception
     {
-        SecretKeySpec skeySpec = new SecretKeySpec(decoder.decode(PLAINTEXT_DATA_KEY), "AES");
-        Cipher cipher = Cipher.getInstance("AES");
-        cipher.init(Cipher.ENCRYPT_MODE, skeySpec);
+        Cipher cipher = AthenaUDFHandler.getCipher(Cipher.ENCRYPT_MODE, decoder.decode(PLAINTEXT_DATA_KEY.getBytes()), AthenaUDFHandler.getGCMSpecEncryption());
 
         String expected = "abcdef";
-        String encryptedString = new String(encoder.encode(cipher.doFinal(expected.getBytes())));
+        byte[] encryptedString = cipher.doFinal(expected.getBytes(StandardCharsets.UTF_8));
+        ByteBuffer byteBuffer = ByteBuffer.allocate(AthenaUDFHandler.GCM_IV_LENGTH + encryptedString.length);
+        byteBuffer.put(cipher.getIV());
+        byteBuffer.put(encryptedString);
 
-        String result = athenaUDFHandler.decrypt(encryptedString, DUMMY_SECRET_NAME);
-
-        assertEquals(expected, result);
-    }
-
-    @Test
-    public void testKmsEncryption() throws Exception
```

Comment:
This test basically became unworkable, as there was no way to pass in the IV to `encrypt` so that they would yield the same result without some horrific monkeypatching (which I'm not entirely sure is possible).

Reply:
Good idea, I'll take a look -- one thing I don't really want to do is leave secure random generation up to the user here, we should just do correct defaults in this case. I'm not yet sure if what you're saying implies this, rather just stating a principle I'm following.